credentials/tls_ext_test.go

```diff
@@ -236,3 +238,99 @@ func (s) TestTLS_CipherSuitesOverridable(t *testing.T) {
 		t.Fatalf("EmptyCall err = %v; want <nil>", err)
 	}
 }
+
+// TestTLS_DisabledALPN tests the behaviour of a gRPC client when connecting to
+// a server that doesn't support ALPN.
+func (s) TestTLS_DisabledALPN(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	initialVal := envconfig.EnforceALPNEnabled
+	defer func() {
+		envconfig.EnforceALPNEnabled = initialVal
+	}()
+
+	// Start a non gRPC TLS server.
+	config := &tls.Config{
+		Certificates: []tls.Certificate{serverCert},
+		NextProtos:   []string{}, // Empty list indicates ALPN is disabled.
+	}
+	listner, err := tls.Listen("tcp", ":0", config)
+	if err != nil {
+		t.Fatalf("Error starting TLS server: %v", err)
+	}
+	defer listner.Close()
+
+	tests := []struct {
+		description            string
+		alpnEnforced           bool
+		wantErrMatchPattern    string
+		wantErrNonMatchPattern string
+	}{
+		{
+			description:         "enforced",
```

Comment:
nit: description of tests have more details. For example, its hard to understand which one of these is a success test for handshake. Alternatively, can add comments as well explaining the intent

go/go-test-examples#descriptions

Reply:
This field is actually the test name and not a description. Renamed the field accordingly. Based on [the Google style guide](https://google.github.io/styleguide/go/decisions.html#subtest-names), the test names should be similar to identifiers like function names to help filter and identify tests instead of a prose description.

Reply:
Thanks for renaming? So, we don't need description here?

Reply:
Having a test description is optional.